### PR TITLE
Clean package file, add coverage and lint script

### PR DESCRIPTION
**Pull request by julien**
Hey @egulias,

Cool stuff. 
I just added a few things in the package file, and "cleaned" it.

I have several PR's for code but I thought it would be better to do them separately (feature by feature)
Cheers

### CODE REVIEW

**Comment by julien** [comment]
One other thing. I added an .eslintrc file but I don't really know your code style preferences (tabs/spaces, single/double quotes, etc...) so I'll let you check that out ;)

File: src/lexer.js [hunk 1]
```diff
@@ -78,97 +87,96 @@ function Lexer(options, rules) {
...
-          return true;
-        }
-
-        tokens.push(new t(specialTokens[match], match));
-        return true;
-      }
-      tokens.push(new t(this.generic, match));
-      return true;
-    }, this);
+  // About "privacy" ...
```
Comment (julien) [inline src/lexer.js]: Remember what we were talking about this morning :)

File: src/lexer.js [hunk 2]
```diff
@@ -42,15 +43,23 @@ var specialTokens = {
...
+};
+
+var tokens = [];
+var position = 1;
+var previous = {
+  equals: function (token) {
+    return false;
+  }
+};
+
```
Comment (egulias) [inline src/lexer.js]: shouldn't these one be part of Lexer? they are really internal to it, not global to the context.

File: src/emailValidator.js [hunk 3]
```diff
@@ -1,19 +1,17 @@
...
-      return false;
-    }
+function EmailValidator() { }
 
-    return true;
-  };
-}
+EmailValidator.prototype.isValid = function (email, strict) {
+  try {
+    parser.parser.parse(email);
```
Comment (egulias) [inline src/emailValidator.js]: Here should be `parser.parse(email)` as per line 3

Comment (egulias) [inline src/emailValidator.js]: Here should be `parser.parse(email)` as per line 3

Comment (julien) [inline src/emailValidator.js]: Yup

File: src/lexer.js (hunk 2 above)
Comment (julien) [inline src/lexer.js]: Ok makes sense

**Pull request merged**